python/tests/image/test_imageIO.py

```diff
@@ -72,71 +74,59 @@ def tearDownClass(cls):
         super(TestReadImages, cls).tearDownClass()
         cls.binaryFilesMock = None
 
-    def test_decodeImage(self):
-        badImg = imageIO._decodeImage(b"xxx")
-        self.assertIsNone(badImg)
-        imgRow = imageIO._decodeImage(pngData)
-        self.assertIsNotNone(imgRow)
-        self.assertEqual(len(imgRow), len(imageIO.imageSchema.names))
-        for n in imageIO.imageSchema.names:
-            imgRow[n]
-
     def test_resize(self):
         imgAsRow = imageIO.imageArrayToStruct(array)
-        smaller = imageIO._resizeFunction([4, 5])
+        imgAsPIL = PIL.Image.fromarray(obj=imageIO._reverseChannels(array)).resize((5,4))
+        smallerAry = imageIO._reverseChannels(np.asarray(imgAsPIL))
+        smaller = imageIO.createResizeImageUDF([4, 5]).func
         smallerImg = smaller(imgAsRow)
-        for n in imageIO.imageSchema.names:
+        np.testing.assert_array_equal(smallerAry,imageIO.imageStructToArray(smallerImg))
+        for n in ImageSchema.imageSchema['image'].dataType.names:
             smallerImg[n]
         self.assertEqual(smallerImg.height, 4)
         self.assertEqual(smallerImg.width, 5)
-
-        sameImage = imageIO._resizeFunction([imgAsRow.height, imgAsRow.width])(imgAsRow)
-        self.assertEqual(sameImage, sameImage)
-
-        self.assertRaises(ValueError, imageIO._resizeFunction, [1, 2, 3])
-
-    def test_imageArrayToStruct(self):
-        SparkMode = imageIO.SparkMode
-        # Check converting with matching types
-        height, width, chan = array.shape
-        imgAsStruct = imageIO.imageArrayToStruct(array)
-        self.assertEqual(imgAsStruct.height, height)
-        self.assertEqual(imgAsStruct.width, width)
-        self.assertEqual(imgAsStruct.data, array.tobytes())
-
-        # Check casting
-        imgAsStruct = imageIO.imageArrayToStruct(array, SparkMode.RGB_FLOAT32)
-        self.assertEqual(imgAsStruct.height, height)
-        self.assertEqual(imgAsStruct.width, width)
-        self.assertEqual(len(imgAsStruct.data), array.size * 4)
-
-        # Check channel mismatch
```

Comment:
do we want to keep these assertRaises tests?

Reply:
The assert raises assertions are not needed anymore. Data type is inferred from the array type instead of being passed as an argument. there is no invalid conversion any more.